modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -863,6 +872,8 @@ public CompletableFuture<Ignite> start(Path configPath) {
                     }, startupExecutor)
                     .thenRunAsync(() -> {
                         try {
+                            //Enable REST component on start complete.
+                            restComponent.enable();
                             // Transfer the node to the STARTED state.
                             lifecycleManager.onStartComplete();
```

Comment:
btw, lifecycleManager can be a good candidate to manage the rest state there or notify the rest component.